src/lib/eslint/TypescriptEslintStrategy.ts

```diff
@@ -0,0 +1,232 @@
+		"node_modules/**"
+	],
+	"useEslintrc": false // TODO derive from existing eslintrc if found and desired
+
+};
+
+const TS_CONFIG = "tsconfig.json";
+
+export class TypescriptEslintStrategy implements EslintStrategy {
+	private static ENGINE_NAME = "eslint-typescript";
+	private static LANGUAGE = ["typescript"];
+
+	private initialized: boolean;
+	protected logger: Logger;
+	private fileHandler: FileHandler;
+	private config: Config;
+	private outputProcessor: OutputProcessor;
+
+	public async init(): Promise<void> {
+		if (this.initialized) {
+			return;
+		}
+		this.logger = await Logger.child(this.getName());
+		this.fileHandler = new FileHandler();
+		this.config = await Controller.getConfig();
+		this.outputProcessor = await OutputProcessor.create({});
+		this.initialized = true;
+	}
+
+	isEnabled(): boolean {
+		return this.config.isEngineEnabled(this.getName());
+	}
+
+	getName(): string {
+		return TypescriptEslintStrategy.ENGINE_NAME;
+	}
+
+	isRuleKeySupported(): boolean {
+		// Javascript rules written for only eslint work for typescript as well,
+		// though they require the parser information that is specific to typescript.
+		return true;
+	}
+
+	getCatalogConfig(): Record<string, any> {
+		return ES_PLUS_TS_CONFIG;
+	}
+
+	getLanguage(): string[] {
+		return TypescriptEslintStrategy.LANGUAGE;
+	}
+
+	async getTargetPatterns(target?: string): Promise<string[]> {
+		const engineConfig = this.config.getEngineConfig(this.getName());
+
+		
+		// Find the typescript config file, if any
+		const tsconfigPath = await this.findTsconfig(target);
+
+		const targetPatterns: string[] = [];
+		if (tsconfigPath) {
+			const tsconfig: TSConfig = JSON.parse(await this.fileHandler.readFile(tsconfigPath));
+
+			// Found a tsconfig.  Load up its patterns.
+			if (tsconfig.include) {
+				targetPatterns.push(...tsconfig.include);
+			}
+
+			if (tsconfig.exclude) {
+				// Negate the exclude pattern (because that's how we like it)
+				targetPatterns.push(...tsconfig.exclude.map(e => "!" + e));
+			} else if (tsconfig.compilerOptions && tsconfig.compilerOptions.outDir) {
+				// TS likes to auto-exclude the outDir but only if exclude is not specified.
+				targetPatterns.push("!" + path.join(tsconfig.compilerOptions.outDir, "**"));
+			} else if (tsconfig.compilerOptions && tsconfig.compilerOptions.outFile) {
+				// Same reasoning as outDir
+				targetPatterns.push("!" + tsconfig.compilerOptions.outFile);
+			}
+
+			if (tsconfig.files) {
+				targetPatterns.push(...tsconfig.files);
+			}
+		}
+
+		// Join forces.  TODO or maybe not?  Should we just rely only on tsconfig if provided?
+		return engineConfig.targetPatterns.concat(targetPatterns);
+	}
+
+	filterUnsupportedPaths(paths: string[]): string[] {
+		const filteredPaths = paths.filter(p => p.endsWith(".ts"));
+		this.logger.trace(`Input paths: ${paths}, Filtered paths: ${filteredPaths}`);
+		return filteredPaths;
+	}
+
+	async getRunConfig(target?: string): Promise<Record<string, any>> {
```

Comment:
We should come this far only when are sure we have `.ts` files to scan. If we still can't find a tsconfig.json file, throw an error and let user provide us a tsconfig file.